services/billing.py

```diff
@@ -432,15 +432,33 @@ def update_email_address(self, owner: Owner, email_address: str):
     @_log_stripe_error
     def update_billing_address(self, owner: Owner, billing_address):
         log.info(f"Stripe update billing address for owner {owner.ownerid}")
-        if owner.stripe_subscription_id is None:
+        if owner.stripe_customer_id is None:
             log.info(
-                f"stripe_subscription_id is None, cannot update billing address for owner {owner.ownerid}"
+                f"stripe_customer_id is None, cannot update default billing address for owner {owner.ownerid}"
             )
             return None
-        stripe.Customer.modify(owner.stripe_customer_id, address=billing_address)
-        log.info(
-            f"Stripe successfully updated billing address for owner {owner.ownerid} by user #{self.requesting_user.ownerid}"
-        )
+
+        try:
```

Comment:
This logic makes sense, but I find it weird we were originally doing it w/ subscription an not customer_id; can you think why that was the case?

Reply:
This was on me and my lack of understanding of where the address / billing details lay among the stripe objects. After actually doing the testing locally and gaining more context on what exactly we are changing and where made me realize this update had nothing to do with the stripe subscription and everything to do with the customer.

Namely, you can have a default payment method even without having a subscription, since that's tied to the customer through payment methods ultimately.